index.js

```diff
@@ -11,6 +11,21 @@ const changePrototype = (to, from) => {
 	Object.setPrototypeOf(to, fromPrototype);
 };
 
+const wrappedToString = (fromBody, name) => {
+	const withToName = name === '' ? '' : `with ${name}() `;
+	return `/* Wrapped ${withToName}*/\n${fromBody}`;
+};
+
+const changeToString = (to, from, name) => {
+	const value = wrappedToString.bind(null, from.toString(), name.trim());
```

Comment:
We call `from.toString()` early as opposed to lazily. We also use `bind()` instead of a closure. This is to ensure `from` can be garbage collected. Otherwise `to.toString()` closure would contain a reference to `from`.

Also calling `from.toString()` early allows caching it in case `to.toString()` is called several times. I'm not sure whether that's already done by v8 under the hood though.